lib/tasks/migration.rake

```diff
@@ -854,6 +871,45 @@ namespace :migration do
 
   end
 
+  def migrate_google_stats(file_dir)
+
+    MigrationLogger.info " +++++++ START: migrate google stats #{file_dir} +++++++ "
+    allfiles = Dir.glob(file_dir+"*.txt")
+    filecount = allfiles.select { |file| File.file?(file) }.count
+    MigrationLogger.info "Files to process: " + filecount.to_s
+    allfiles.sort.each_with_index do |file, thisfile|
+      start_time = Time.now
+      MigrationLogger.info "Processing the file #{file} (#{thisfile + 1} of #{filecount})"
+      index = file.rindex('/')
+      id = file[index+1..-1]
+      id.slice! ".txt"
+     
+      begin 
+        @generic_file = GenericFile.find(id)
+        if @generic_file == nil
+          MigrationLogger.info "Generic file not found: #{id}"
+        else
+          @generic_file.add_file(File.open(file), path: 'gastats', original_name: file, mime_type: 'texl/xml')
```

Comment:
Could we add a check here, to prevent remigrating the stats if they're already there? We're still running migration jobs and we may well process the same items more than once.